prefect-v1-legacy/devops/monitoring/README.md

```diff
@@ -0,0 +1,345 @@
+helm upgrade prefect-prometheus ./prefect-prometheus
+```
+
+Optionally, a `.yaml` can be created to isolate the changes listed in Step 2 and passed as an argument. The arguments passed in this fashion will supersede those defined in `values.yaml`.
+This can be useful if you have separate Prefect Servers you’d like to apply (using separate GraphQL endpoints) or different environments entirely.
+
+```yaml
+vim custom_values.yaml
+image:
+  repository: **chaboy/prefect-prometheus-exporter**
+  tag: **"1.2.3-amd64"**
+
+service:
+  port: **8000**
+  targetPort: **8000**
+
+graphqlEndpoint: https://apollo.prefect-installation.com
+```
+
+This override of `custom_values.yaml` can then be applied:
+
+```yaml
+helm upgrade prefect-prometheus ./prefect-prometheus -f custom_values.yaml
+```
+
+## 4 - Validating Prometheus
+
+This step is solely focused on Prometheus configuration to observe the prefect-prometheus-exporter.
+
+Access the Prometheus service by either port-forwarding directly into the cluster if the service is not exposed. This requires your `KUBECONFIG` to be set to the cluster, assuming Prometheus is running inside the cluster.
+
+```bash
+#services/prometheus-kube-prometheus - services is the definition you want to forward
+#prometheus-kube-prometheus is the specific name of the service you wish to forward
+#This service is listening on :9090 - we are forwarding our own 9090 traffic to it
+#-n prometheus is for the "prometheus" namespace - this might be different depending on where prometheus is installed
+kubectl port-forward services/prometheus-kube-prometheus-prometheus 9090:9090 -n prometheus
+```
+
+You can then access the page via [localhost:9090](http://localhost:9090).
+
+Alternatively, if the service has an IP address and is exposed (either internally or externally) you can access that *ip:port* address directly.
+
+Once in the Prometheus configuration, you can validate that the exporter is visible by navigating to “Status → Targets”. We can see in the image under “Targets” that prefect-prometheus is **(1/1 up)**, indicating Prometheus is aware and pulling metrics from this configuration.
+
+![Untitled](https://github.com/PrefectHQ/prefect-recipes/blob/prom-monitoring/monitoring/imgs/prometheus.png)
+
+![Untitled](https://github.com/PrefectHQ/prefect-recipes/blob/prom-monitoring/monitoring/imgs/prom%20target.png)
+
+## 5 - Accessing the Grafana Dashboard
+
+This is identical to Step 4. Here we are substituting the Prometheus Service and Endpoint for Grafana:
+
+```bash
+# Via port-forwarding
+kubectl port-forward services/prometheus-grafana 3000:3000 -n prometheus
+```
+
+Then we can access Grafana on  [localhost:3000](http://localhost:3000). Alternatively, we can reach the endpoint directly through `ip:port`.
+
+Logins for Grafana should exist in 1Password. Once logged in, you’ll be presented with a list of recently viewed dashboards. If this is the first time logging in, verify Grafana is pulling in metrics from Prometheus:
```

Comment:
Might be worth removing this as I believe it's specific to Exscientia?

Reply:
Updated and removed, thanks for the review!